src/Extension/FluentExtension.php

```diff
@@ -650,55 +660,42 @@ public function onAfterWrite(): void
         }
 
         // Get list of all localised instances of this model and duplicate relations if needed (if current one has it)
-        $locales = $owner->getLocaleInstances();
-        $objectsToWrite = [];
+        foreach ($this->owner->Locales() as $recordLocale) {
+            // Skip locales this record isn't localised in
+            if (!$recordLocale->IsDraft()) {
+                continue;
+            }
 
-        /** @var Locale $locale */
-        foreach ($locales as $locale) {
-            if ($locale->Locale === $currentLocale) {
+            $locale = $recordLocale->getLocaleObject();
```

Comment:
[nitpick] This looks like unused variable.
```suggestion
```

Reply:
ah good spotting.